### PR TITLE
[Book_Store] order functionality

### PR DESCRIPTION
**Pull request by fedorovychh**
Added Order functionality:
- controller
- Order entity
- OrderItem entity
- services
- dtos
- repositories
- added liquibase changeset
- mappers

### CODE REVIEW

**Review by dima-semeniuk — APPROVED** [review]
Great job! I have nothing to fix)

File: src/main/java/com/app/bookstore/model/Order.java
Comment (d1sam) [inline src/main/java/com/app/bookstore/model/Order.java:1]: add `nullable = false` to fields

File: src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    @Override
+    public List<OrderResponseDto> getAll(Long id) {
+        List<Order> orderList = orderRepository.findAllById(id);
+        return orderList.stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderResponseDto updateStatus(Long id, UpdateOrderRequestDto requestDto) {
```
Comment (d1sam) [inline src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java:60]:
```suggestion
    @Transactional
    public OrderResponseDto updateStatus(Long id, UpdateOrderRequestDto requestDto) {

```

File: src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(User user, OrderRequestDto requestDto) {
```
Comment (d1sam) [inline src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java]:
```suggestion
    public OrderResponseDto placeOrder(Long userId, OrderRequestDto requestDto) {

```

File: src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    @Override
+    public OrderResponseDto updateStatus(Long id, UpdateOrderRequestDto requestDto) {
+        Order order = orderRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order by id: " + id)
+        );
+        order.setStatus(requestDto.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    private Order createOrder(User user, ShoppingCart shoppingCart, OrderRequestDto requestDto) {
```
Comment (d1sam) [inline src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java]:
```suggestion
    private Order createOrder(Long userId, ShoppingCart shoppingCart, OrderRequestDto requestDto) {

```

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but there are some changes should be done!

File: src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java (hunk 3 above)
Comment (fedorovychh) [reply src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java]: I set user to the order in this method, so I can't change it

File: src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java (hunk 2 above)
Comment (fedorovychh) [reply src/main/java/com/app/bookstore/service/order/OrderServiceImpl.java]: I set user to the order in this method, so I can't change it

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**